docs/tutorials/complete_applications/complete_application/step_04/step_04.md

```diff
@@ -0,0 +1,181 @@
+
+- **Predictions Chart**: `<|{predictions_dataset}|chart|...|>`
+
+A chart that displays historical values and the predicted values obtained from machine learning and baseline methods. It shows how well the predictions align with the historical data.
+
+- **Data Node Exploration**: `<|{data_node}|data_node_selector|> <|{data_node}|data_node_selector|>`
+
+This is where the detailed information and history about the selected data node is presented. Depending on the nature of the data node, this could display raw data in a tabular format, visualizations, texts, or dates. If the format allows it, the user can directly write new values in the data node.
+
+
+# Python Code (pages/scenario/scenario.py)
+
+The Python code initializes and manages the state of the Scenario Page.
+
+```python
+from taipy.gui import Markdown, notify
+import datetime as dt
+import pandas as pd
+
+
+scenario = None
+data_node = None
+day = dt.datetime(2021, 7, 26)
+n_predictions = 40
+max_capacity = 200
+predictions_dataset = {"Date":[0], 
+                       "Predicted values ML":[0],
+                       "Predicted values Baseline":[0],
+                       "Historical values":[0]}
+
+def submission_change(state, submittable, details: dict):
+    if details['submission_status'] == 'COMPLETED':
+        notify(state, "success", 'Scenario completed!')
+        state['scenario'].on_change(state, 'scenario', state.scenario)
+    else:
+        notify(state, "error", 'Something went wrong!')
+
+def save(state):
+    print("Saving scenario...")
+    # Get the currently selected scenario
+
+    # Conversion to the right format
+    state_day = dt.datetime(state.day.year, state.day.month, state.day.day)
+
+    # Change the default parameters by writing in the Data Nodes
+    state.scenario.day.write(state_day)
+    state.scenario.n_predictions.write(int(state.n_predictions))
+    state.scenario.max_capacity.write(int(state.max_capacity))
+    notify(state, "success", "Saved!")
+    
+
+def on_change(state, var_name, var_value):
+    if var_name == "scenario" and var_value:
+        state.day = state.scenario.day.read()
+        state.n_predictions = state.scenario.n_predictions.read()
+        state.max_capacity = state.scenario.max_capacity.read()
+        
+        if state.scenario.full_predictions.is_ready_for_reading:
+            state.predictions_dataset = state.scenario.full_predictions.read()
+        else:
+            state.predictions_dataset = predictions_dataset
+
+
+
+scenario_page = Markdown("pages/scenario/scenario.md")
+```
+
+
+It includes the following components:
+
+- **Global Variables**:
+
+The global variables *scenario*, *data_node*, *day*, *n_predictions*, *max_capacity*, and *predictions_dataset* are defined. These variables store the initial state of the application.
+
+- **Save Function**:
+
+The `save` function is in charge of preserving the current scenario state. 
```

Comment:
Should we use the term callback instead of function? 
```suggestion
The `save` function is used as a callback. It is in charge of preserving the current scenario state. 
```
Then once callback is defined, we can use callback in place of function. What do you think?